gdspy/gdsiiformat.py

```diff
@@ -213,6 +213,8 @@ def gdsii_hash(filename, engine=None):
             start = pos + 28
         elif rec == 0x0700:
             contents.append(data[start:pos])
+        elif size == 0:
```

Comment:
Before breaking, the function should also raise a warning, because this is not a strictly valid GDSII file and it could indicate some sort of corruption. Something like:

```python
warnings.warn(
    "[GDSPY] Zero-length record found in {0} at position {1}. "
    "Skipping the remaining of the file contents.".format(filename, pos),
    stacklevel=2,
)
```

Reply:
sounds good. i made the change. FYI when looking at the file size, it looked like after the padding it was an even multiple of 1024, so i'm guessing maybe the program which made this was writing in fixed length chunks and continued to write all the hanging nulls after the file content ended (?). it was a big-name EDA tool, but they don't always live up to their cost 😆